modules/metrics/src/test/java/org/apache/ignite/internal/metrics/AbstractScalarMetricsTest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.metrics;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.apache.ignite.internal.metrics.Metric;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Abstract test for scalar metrics.
+ *
+ * @param <M> Metric type.
+ * @param <V> Type of metric value.
+ */
+public abstract class AbstractScalarMetricsTest<M extends Metric, V> {
+    @Test
+    public void metricTest() {
```

Comment:
Again, could you separate it to smaller logical tests, it will be great to see, which part of logic failed from the test name on TC, for example.